### ISSUE TITLE
OIDC/SSO support in general?

### ISSUE THREAD

**Issue by JaneJeon** [sissbruecker/linkding#177]
Would be good to see OIDC or SAML support in LD so we don't have to manually create users and have an extra set of user/pass to memorize.

**Comment by Tuphal** [comment]
this.
Support for AuthProxy like Keycloak or Authleia would be also perfect. And/or LDAP support

**Comment by sissbruecker** [comment]
Since linkding is intended for self-hosting scenarios, and most deployments probably have just a single user, I don't see a lot of value in this. Not to mention that someone would have to host their own auth provider in that scenario, which is not impossible (seeing the feedback here), but still very unlikely.

I don't think I would want to move support for this into the codebase, however maybe someone can come up with a solution to mount custom settings into the Docker container to make Django use an authentication provider, rather than use the built-in authentication mechanism.

**Comment by grantemsley** [comment]
I too would love SSO integration. My selfhosted stuff is used by many of my family and friends, most of which use LDAP to login, though I'm also looking into Authleia.

Since you use django, it shouldn't be too difficult to add some of the SSO modules that already exist for it. Ideally the docker container would have all the necessary modules available but not configured, and those who want it could configure it themselves.

Mounting custom settings in docker isn't too difficult. I've done it for this project: https://github.com/grantemsley/captiveportal
using instructions I found at http://www.robgolding.com/blog/2010/05/03/extending-settings-variables-with-local_settings-py-in-django/

Basically you create a local_settings.py file in the data folder that gets symlinked to the app folder on startup. Then users can edit any django settings they want.  If you're using django's built in user management, all you should have to do is add a couple modules like django-auth-ldap and djangosaml2 and setup that local_settings.py file to be symlinked and read so users can configure the rest.

**Comment by sama8** [comment]
If relevant -> [SAML Jackson](https://github.com/boxyhq/jackson)

**Comment by bachya** [comment]
For anyone wanting to take a crack at this, it looks like the Authelia community has instructions on how to work with a Django app: https://www.authelia.com/docs/community/django-integration.html

**Comment by max-tet** [comment]
I believe an proxy auth option would be useful in particular, even if the app is only used by a single user. Hosting multiple apps on a homer server and behind a reverse proxy that does authentication for you is common and very convenient.

So +1 for a proxy auth option!

**Comment by sissbruecker** [comment]
Added support for authentication proxies in https://github.com/sissbruecker/linkding/pull/321

**Comment by wachtell** [comment]
> For anyone wanting to take a crack at this, it looks like the Authelia community has instructions on how to work with a Django app: https://www.authelia.com/docs/community/django-integration.html

Looks like the link doesn't work :-)

**Comment by huyz** [comment]
@wachtell https://linkding.link/options/#ld_enable_auth_proxy suffices
